spec-tests/src/test/java/org/candlepin/spec/jobs/InactiveConsumerCleanerJobSpecTest.java

```diff
@@ -136,6 +138,64 @@ public void shouldDeleteInactiveConsumers() {
         compareConsumers(activeConsumer2, consumerApi.getConsumer(activeConsumer2.getUuid()));
     }
 
+    @Test
+    public void shouldUpdateDeletedConsumerWhenUuidReused() {
+        final String uuid = UUID.randomUUID().toString();
+
+        Instant inactiveTime = Instant.now()
+            .minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS + 10, ChronoUnit.DAYS);
+
+        ConsumerDTO c1 = Consumers.random(owner)
+            .uuid(uuid)
+            .lastCheckin(inactiveTime.atOffset(ZoneOffset.UTC));
+        c1 = consumerApi.createConsumer(c1, null, owner.getKey(), null, false);
+
+        // Delete Consumer1 (simulate previous deletion in the past)
+        consumerApi.deleteConsumer(c1.getUuid());
+
+        // Capture the first deletion record (must exist now)
+        List<DeletedConsumerDTO> afterFirstDeletion = deletedConsumerApi
+            .listByDate(c1.getCreated(), 1, 200, "desc", "created");
+        List<DeletedConsumerDTO> firstDeletionForUuid = afterFirstDeletion.stream()
+            .filter(dc -> uuid.equals(dc.getConsumerUuid()))
+            .toList();
+
+        assertThat(firstDeletionForUuid)
+            .hasSize(1);
+
+        OffsetDateTime firstDeletedCreated = firstDeletionForUuid.get(0).getCreated();
+
+        // create Consumer2 with the SAME UUID, make it inactive
+        ConsumerDTO c2 = Consumers.random(owner)
+            .uuid(uuid)
+            .lastCheckin(inactiveTime.atOffset(ZoneOffset.UTC));
+        c2 = consumerApi.createConsumer(c2, null, owner.getKey(), null, false);
+
+        String jobId = jobsClient.scheduleJob(JOB_KEY).getId();
+        AsyncJobStatusDTO status = jobsClient.waitForJob(jobId);
+        assertEquals("FINISHED", status.getState());
```

Comment:
You can use `assertThatJob(status).isFinished();` here so that we are not hard coding the "FINISHED" status

Reply:
done